File: Java-Module/src/main/java/com/sim/javamodule/domain/product/service/dto/ProductInfoResponse.java
package com.sim.javamodule.domain.product.service.dto;

import com.sim.javamodule.domain.product.domain.Product;
import lombok.Getter;

public record ProductInfoResponse(String name, Long cost, Long quantity) {
    public static ProductInfoResponse mapProductToProductInfoResponse(final Product product){
        return new ProductInfoResponse(product.getName(), product.getCost(), product.getQuantity());
    }
}

File: Kotlin-Module/src/main/kotlin/com/sim/kotlin/domain/product/service/dto/ProductInfoResponse.kt
package com.sim.kotlin.domain.product.service.dto

import com.sim.kotlin.domain.product.domain.Product

data class ProductInfoResponse(
    val name: String,
    val cost: Long,
    val quantity: Long,
) {
    companion object{
        fun mapProductToProductInfoResponse(product: Product) : ProductInfoResponse{
            return ProductInfoResponse(product.name, product.cost, product.quantity)
        }
//        fun mapProductToProductInfoResponse(product: Product)
//            = ProductInfoResponse(product.name, product.cost, product.quantity)

    }
}